src/test/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceIT.java

```diff
@@ -177,4 +166,58 @@ public final void testListInstitution(final Vertx aVertx, final VertxTestContext
         }).onFailure(aContext::failNow);
     }
 
+    /**
+     * Tests deleting institution from db.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testDeleteInstitution(final Vertx aVertx, final VertxTestContext aContext)
+            throws AddressException, MalformedURLException, NumberParseException {
+        final Institution toDelete = TestUtils.getRandomInstitution();
+        myScheduleStoreProxy.addInstitution(toDelete).onSuccess(newID -> {
+            myScheduleStoreProxy.removeInstitution(newID).onSuccess(result -> {
+                myScheduleStoreProxy.getInstitution(newID).onFailure(details -> {
+                    final ServiceException error = (ServiceException) details;
+
+                    aContext.verify(() -> {
+                        assertEquals(Error.NOT_FOUND.ordinal(), error.failureCode());
+                        assertTrue(error.getMessage().contains(String.valueOf(newID)));
+                    }).completeNow();
+                }).onSuccess(select -> {
+                    aContext.failNow("delete failed");
+                });
+            });
```

Comment:
the call to `removeInstitution` should probably have a failure handler too

Reply:
Added fail handler